src/filters/dates/relative/relativeDates.ts

```diff
@@ -0,0 +1,74 @@
+import { ParsedMatchSchema } from 'serina.schema';
+import { DateTime } from 'luxon';
+import { DATES } from '../dates.constants';
+import RELATIVE_DATES, { RELATIVE_ADVERB } from './relativeDates.constants';
+import { parseMatches, contains, remove, matchPattern } from 'utils';
+
+export default class RelativeDates {
+    static parseText(text: string): ParsedMatchSchema[] {
+        const pattern = `(${DATES.FILLER_WORDS})?${RELATIVE_DATES.ANY}`;
+        const matches = matchPattern(text, pattern, false);
+
+        if (!matches) return null;
+
+        return matches.map(match => {
+            const dateTimeObj = this.convertMatchToDateObj(match);
+            return parseMatches(text, match, dateTimeObj);
+        });
+    }
+
+    static timeUnitToString(matchAgainst: string): string {
+        for (const key in RELATIVE_DATES.TIME_UNITS) {
+            if (contains(matchAgainst, RELATIVE_DATES.TIME_UNITS[key])) {
+                return key;
+            }
+        }
+        return null;
+    }
+
+    static convertRelativeAdverbToObj(relativeDateStr: string): DateTime {
+        if (contains(relativeDateStr, RELATIVE_ADVERB.TODAY)) {
+            return DateTime.utc();
+        } else if (contains(relativeDateStr, RELATIVE_ADVERB.TOMORROW)) {
+            return DateTime.utc().plus({ days: 1 });
+        }
+        return null;
+    }
+
+    static getNext(unit): DateTime {
+        return DateTime.utc()
+            .plus({ [unit]: 1 })
+            .startOf(unit)
+            .endOf('day');
+    }
+
+    static convertRelativeExpressionToObj(expression: string): DateTime {
+        const unit = this.timeUnitToString(matchPattern(expression, RELATIVE_DATES.TIME_UNITS.ANY)[0]);
+        const period = remove(expression, unit);
+        let quantity;
+        // Multiplication is commutative, we are going to take a quarter of whatever time unit we are going to get.
+        if (contains(period, RELATIVE_DATES.VERBAL_QUANTIFIERS.ONE)) {
+            quantity = 1;
+        } else if (contains(period, RELATIVE_DATES.VERBAL_QUANTIFIERS.NEXT)) {
+            return this.getNext(unit);
+        } else {
+            quantity = parseInt(period, 10);
+        }
+        if (contains(unit, RELATIVE_DATES.TIME_UNITS.ANY)) {
+            return DateTime.utc().plus({ [unit]: quantity });
+        }
+        return null;
```

Comment:
Not sure how to handle this and the other return nulls inside helper methods, at the moment they will never return null, because the if conditions are exhaustive, i.e. the function is only called if the pattern is matched, and the function goes through all the possible pattern matches under current rules. It is necessary to have the return null though due to methods having to return the given type. The workaround would be to have the current last condition as default return (in this case line 58 would not have the if condition. However, this IMO reduces the clarify and might lead to bugs in the future when we add more possibilities. Thoughts?

Reply:
Have no idea what you are trying to say here lol

Reply:
Codecov is complaining that this null check is never hit. The reason it is never hit is that at the moment, the above condition is always true (otherwise this function would not be called, as time units are mandatory for this expression to match).

However, the null check is necessary, because the compiler does not know that the if condition is always met and complains if it isn't.